hbase_storage/src/main/java/com/splicemachine/access/client/SkeletonClientSideRegionScanner.java

```diff
@@ -141,10 +143,44 @@ public boolean isFilterDone() throws IOException{
         return scanner.isFilterDone();
     }
 
+    private void fetchNextAndGiveBuffer(List<Cell> result) throws IOException {
+        assert result.isEmpty();
+        List<Cell> nextResult = new ArrayList<>();
+        boolean response = nextMerged(nextResult);
+        if (matchingFamily(nextResult, ClientRegionConstants.FLUSH)) {
+            // A flush should be returned before a potential partial result in the buffer
```

Comment:
I don't understand this comment, does it mean that we have the full nextResult or could it be partial?

Reply:
Typically, when we return from this function, we have managed to put the buffer in `result` and we've gotten a new buffer.
If we get a flush token, we cannot return the current buffer, because it may contain partial results. Those partial results would never be completed since nextRaw would return FLUSH.
So we need to return the flush before.